File: docs/replace-test-titles.py
from bs4 import BeautifulSoup
import re

def replace_test_links(html_content):
    soup = BeautifulSoup(html_content, 'html.parser')
    pattern = re.compile(r'docs/\d{4}-\d{2}-\d{2}-(.+)\.html')

    for a_tag in soup.find_all('a'):
        if a_tag.string == 'test' and a_tag.has_attr('href'):
            match = pattern.search(a_tag['href'])
            if match:
                term = match.group(1)
                a_tag.string.replace_with(term)

    return str(soup)

# Example usage:
if __name__ == '__main__':
    import sys
    from pathlib import Path

    if len(sys.argv) != 2:
        print("Usage: python replace_link_titles.py <path_to_html_file>")
        sys.exit(1)

    html_path = Path(sys.argv[1])
    html_content = html_path.read_text(encoding='utf-8')
    updated_html = replace_test_links(html_content)
    html_path.write_text(updated_html, encoding='utf-8')
    print(f"Updated file: {html_path}")

File: docs/replace-test-titles.rb
# replace test titles

require 'nokogiri'

def replace_test_links(html)
  doc = Nokogiri::HTML(html)

  doc.css('a').each do |a|
    href = a['href']
    next unless a.text == 'test' && href

    # Match pattern like: docs/2025-07-01-affinity.html
    if href =~ %r{docs/\d{4}-\d{2}-\d{2}-(.+)\.html}
      term = $1
      a.content = term
    end
  end

  doc.to_html
end

# Command-line usage
if __FILE__ == $0
  unless ARGV.size == 1
    puts "Usage: ruby replace_link_titles.rb <path_to_html_file>"
    exit 1
  end

  file_path = ARGV[0]
  html = File.read(file_path)
  updated_html = replace_test_links(html)
  File.write(file_path, updated_html)
  puts "Updated file: #{file_path}"
end
